pipeline/transform/matches.py

```diff
@@ -24,7 +30,10 @@ def convert_matches_from_df(df: pd.DataFrame) -> List[Match]:
     cols = set(df.columns)
     field_names = [f.name for f in dataclasses.fields(Match) if f.name in cols]
     user_dicts = df[field_names].to_dict(orient="records")
-    return [Match(**u) for u in user_dicts]
+    matches = [Match(**u) for u in user_dicts]
+    for match in matches:
+        match.metadata = MatchMetadata(str(match.metadata))
```

Comment:
Why are we casting `match.metadata` to a string here? This is also related to why the unit tests for this pull request are failing.

Reply:
I tried different combinations of types and this was the only thing that worked. In hindsight, this was probably not the right thing to do. Though it is fixed now. :)